### PR TITLE
DockerRun

### PR DESCRIPTION
**Pull request by Eskibear**

### CODE REVIEW

File: PluginsAndFeatures/azure-toolkit-for-eclipse/com.microsoft.azuretools.container/src/main/java/com/microsoft/azuretools/container/DockerRuntime.java [hunk 1]
```diff
@@ -22,151 +22,71 @@
...
-
-    public synchronized String getRegistryPassword() {
-        return registryPassword;
-    }
-
-    public synchronized void setRegistryPassword(String registryPassword) {
-        this.registryPassword = registryPassword;
-    }
+    // project basePath as key
+    private Map<String, DockerHostRunSetting> dockerSettingMap = new ConcurrentHashMap<>();
```
Comment (adashen) [inline PluginsAndFeatures/azure-toolkit-for-eclipse/com.microsoft.azuretools.container/src/main/java/com/microsoft/azuretools/container/DockerRuntime.java]: use one map to hold these two kind of info

File: PluginsAndFeatures/azure-toolkit-for-eclipse/com.microsoft.azuretools.container/src/main/java/com/microsoft/azuretools/container/DockerRuntime.java [hunk 2]
```diff
@@ -22,151 +22,71 @@
...
+        dockerSettingMap.put(key, model);
     }
 
     /**
      * clean running container.
      */
-    public synchronized void cleanRuningContainer() throws DockerException, InterruptedException {
-        if (runningContainerId != null) {
-            DockerClient docker = dockerBuilder.build();
+    public synchronized void cleanRuningContainer(String key)
```
Comment (adashen) [inline PluginsAndFeatures/azure-toolkit-for-eclipse/com.microsoft.azuretools.container/src/main/java/com/microsoft/azuretools/container/DockerRuntime.java]: still need public?

File: PluginsAndFeatures/azure-toolkit-for-eclipse/com.microsoft.azuretools.container/src/main/java/com/microsoft/azuretools/container/ui/common/ContainerSettingComposite.java [hunk 3]
```diff
@@ -72,8 +70,8 @@
     private Label lblStartupFile;
     private Combo cbContainerRegistry;
     private Label lblDockerFile;
-    private Text txtDockerFile;
-    private Button btnDockerFileSelector;
+    private Composite composite;
```
Comment (adashen) [inline PluginsAndFeatures/azure-toolkit-for-eclipse/com.microsoft.azuretools.container/src/main/java/com/microsoft/azuretools/container/ui/common/ContainerSettingComposite.java]: where used composite?

File: PluginsAndFeatures/azure-toolkit-for-eclipse/com.microsoft.azuretools.container/src/main/java/com/microsoft/azuretools/container/ui/common/FileSelector.java [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+import org.eclipse.swt.widgets.Text;
+
+public class FileSelector extends Composite {
+    private Text txtFilePath;
+
+    /**
+     * Create the composite.
+     */
+    public FileSelector(Composite parent, int style, boolean isDir, String btnText, String basePath) {
+        super(parent, style);
```
Comment (adashen) [inline PluginsAndFeatures/azure-toolkit-for-eclipse/com.microsoft.azuretools.container/src/main/java/com/microsoft/azuretools/container/ui/common/FileSelector.java]: parent not null?

File: PluginsAndFeatures/azure-toolkit-for-eclipse/com.microsoft.azuretools.container/src/main/java/com/microsoft/azuretools/container/DockerRuntime.java (hunk 2 above)
Comment (Eskibear) [reply PluginsAndFeatures/azure-toolkit-for-eclipse/com.microsoft.azuretools.container/src/main/java/com/microsoft/azuretools/container/DockerRuntime.java]: yes, might be called before docker run

File: PluginsAndFeatures/azure-toolkit-for-eclipse/com.microsoft.azuretools.container/src/main/java/com/microsoft/azuretools/container/ui/common/FileSelector.java (hunk 4 above)
Comment (Eskibear) [reply PluginsAndFeatures/azure-toolkit-for-eclipse/com.microsoft.azuretools.container/src/main/java/com/microsoft/azuretools/container/ui/common/FileSelector.java]: Parameters:
parent - a widget which will be the parent of the new instance (cannot be null)
style - the style of widget to construct
Throws:
IllegalArgumentException -
ERROR_NULL_ARGUMENT - if the parent is null
SWTException -
ERROR_THREAD_INVALID_ACCESS - if not called from the thread that created the parent

**Pull request merged**